### PR TITLE
fixup default --insecure-registry CIDR address

### PR DESCRIPTION
**Pull request by bacongobbler**
The default CIDR should be 10.0.0.0/24 rather than 10.0.0.1.
Accidental off-by-one error. :)

### CODE REVIEW

**Comment by aaron-prindle** [comment]
@minikube-bot ok to test

**Comment by bacongobbler** [comment]
See #1583 for when this was introduced

**Comment by bacongobbler** [comment]
It also appears that there are a few more bugs like the pods not being in the service's pod selector... More fixes to come in this PR!

**Comment by bacongobbler** [comment]
k, I'm going to manually test these changes before they're ready to merge to make sure I didn't mess anything up this time around.

**Comment by bacongobbler** [comment]
Okay, this now works as intended.

From one pod in the cluster, pushing an image into the registry using the registry service's virtual IP address:

```
--> Pushing 10.0.0.131/draft/py:ab8c330f1325c0b25671841381430498edb98259
The push refers to a repository [10.0.0.131/draft/py]
[...]
e4c17ef2609e: Pushed
ab8c330f1325c0b25671841381430498edb98259: digest: sha256:1877adfe66e955b7f4bdd5fa27cdf908a864a8cf142052f311f19676bc9c8b77 size: 2840
```

Then, with a deployment pointing at the service IP address as the image name:

```
><> k get po py-exasperated-dragon-1723738893-6rw6h -o jsonpath='{.spec.containers[0].image}'
10.0.0.131/draft/py:ab8c330f1325c0b25671841381430498edb98259

><> k get po py-exasperated-dragon-1723738893-6rw6h -o jsonpath='{.status.containerStatuses[0].ready}'
true
```

All good to go :)

**Comment by dlorenc** [comment]
@minikube-bot OK to test

File: cmd/minikube/cmd/start.go [hunk 1]
```diff
@@ -293,7 +293,7 @@ func init() {
 	startCmd.Flags().StringArrayVar(&dockerOpt, "docker-opt", nil, "Specify arbitrary flags to pass to the Docker daemon. (format: key=value)")
 	startCmd.Flags().String(apiServerName, constants.APIServerName, "The apiserver name which is used in the generated certificate for localkube/kubernetes.  This can be used if you want to make the apiserver available from outside the machine")
 	startCmd.Flags().String(dnsDomain, constants.ClusterDNSDomain, "The cluster dns domain name used in the kubernetes cluster")
-	startCmd.Flags().StringSliceVar(&insecureRegistry, "insecure-registry", []string{util.DefaultServiceClusterIP + "/24"}, "Insecure Docker registries to pass to the Docker daemon")
+	startCmd.Flags().StringSliceVar(&insecureRegistry, "insecure-registry", []string{constants.DefaultInsecureRegistry}, "Insecure Docker registries to pass to the Docker daemon")
```
Comment (dlorenc) [inline cmd/minikube/cmd/start.go]: I think you need to reference this as "pkgutil.DefaultInsecureRegistry" (or move the constant definition to pkg/minikube/constants/constants.go)

Comment (bacongobbler) [inline cmd/minikube/cmd/start.go]: Sorry! I fixed this locally but never pushed up the change. Thanks!

File: deploy/addons/registry/registry-rc.yaml [hunk 2]
```diff
@@ -9,11 +9,11 @@ metadata:
 spec:
   replicas: 1
   selector:
-    addonmanager.kubernetes.io/mode: Reconcile
+    kubernetes.io/minikube-addons: registry
   template:
     metadata:
       labels:
-        addonmanager.kubernetes.io/mode: Reconcile
```
Comment (dlorenc) [inline deploy/addons/registry/registry-rc.yaml]: I think you need to keep this Reconcile label here in the template/metadata for the addon manager to pick it up.

Comment (bacongobbler) [inline deploy/addons/registry/registry-rc.yaml]: Is it required for the pod manifest as well? I guess replication controllers don't work the same as deployments. I'll add this back.

**Pull request merged**